### PR TITLE
feat: implement db.rowcount tag across database and cache integrations 

### PR DESCRIPTION
**Pull request by wconti27**
### Description
This PR changes the existing `db.rowcount` tag to be `db.row_count`. It also adds the tag to the mysql and memcached integrations since this is a necessary DB & Cache tag according to new Datadog Span Spec.

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: src/Integrations/Integrations/Memcached/MemcachedIntegration.php [hunk 1]
```diff
@@ -118,56 +118,84 @@ public function init()
...
-                $integration->setServerTags($span, $this);
-                $span->meta['memcached.query'] = $command . ' ' . Obfuscation::toObfuscatedString($args[0]);
-                $span->meta['memcached.server_key'] = $args[0];
+        \DDTrace\trace_method(
+            'Memcached',
+            $command,
+            function (SpanData $span, $args, $retval) use ($integration, $command) {
+                $integration->setCommonData($span, $command);
+                if ($command === 'getByKey') {
+                    $span->metrics[Tag::DB_ROWCOUNT] = isset($retval) && $retval ? 1 : 0;
```
Comment (bwoebi) [inline src/Integrations/Integrations/Memcached/MemcachedIntegration.php:148]:
```suggestion
                    $span->metrics[Tag::DB_ROWCOUNT] = empty($retval) ? 0 : 1;
```

File: src/Integrations/Integrations/Mysqli/MysqliIntegration.php [hunk 2]
```diff
@@ -183,15 +183,20 @@ function (SpanData $span) use ($integration) {
...
+    public function setDefaultAttributes(SpanData $span, $name, $resource, $result = null)
     {
         $span->name = $name;
         $span->resource = $resource;
         $span->type = Type::SQL;
         $span->service = 'mysqli';
         $span->meta[Tag::SPAN_KIND] = 'client';
         $span->meta[Tag::COMPONENT] = MysqliIntegration::NAME;
+
+        if ($result && property_exists($result, 'num_rows')) {
```
Comment (bwoebi) [inline src/Integrations/Integrations/Mysqli/MysqliIntegration.php:197]: Queries without resultset return true, and will throw a type error here:
```suggestion
        if (is_object($result) && property_exists($result, 'num_rows')) {
```

Comment (wconti27) [reply src/Integrations/Integrations/Mysqli/MysqliIntegration.php]: changed this

File: tests/Integrations/PDO/PDOTest.php [hunk 3]
```diff
@@ -72,7 +72,7 @@ public function testCustomPDOPrepareWithStringableStatement()
             )
                 ->setTraceAnalyticsCandidate()
                 ->withExactTags(array_merge($this->baseTags(), [
-                    'db.rowcount' => 1,
+                    'db.row_count' => 1,
```
Comment (morrisonlevi) [inline tests/Integrations/PDO/PDOTest.php:75]: This PR is not consistent with `'db.row_count'` literal vs `Tag::DB_ROWCOUNT` constant.

File: src/api/Tag.php [hunk 4]
```diff
@@ -42,6 +42,9 @@ class Tag
     const ELASTICSEARCH_PARAMS = 'elasticsearch.params';
     const ELASTICSEARCH_URL = 'elasticsearch.url';
 
+    // DB
+    const DB_ROWCOUNT = 'db.row_count';
```
Comment (morrisonlevi) [inline src/api/Tag.php:46]: Also, anyone care about `DB_ROWCOUNT` vs `DB_ROW_COUNT`?

**Review by bwoebi — APPROVED** [review]
Looks good now, thanks @wconti27

**Pull request merged**